emloop/api.py

```diff
@@ -200,25 +208,29 @@ def create_hooks(config: dict, model: Optional[AbstractModel]=None, dataset: Opt
             raise ex
 
     if not training_trace_created:
-        logging.warning('TrainingTrace hook added between hooks. Add it to your config.yaml to suppress this warning.')
+        logging.warning(
+            'TrainingTrace hook added between hooks. Add it to your config.yaml to suppress this warning.')
         hooks.append(TrainingTrace(output_dir=output_dir))
 
     return hooks
 
 
-def create_main_loop(config: dict, output_root: str, restore_from: str=None) -> MainLoop:
+def create_main_loop(config: dict, output_root: str, restore_from: str=None, target_dir: str='') -> MainLoop:
     """
     Creates :py:class:`MainLoop` with model, dataset and hooks according to config.
 
-    :param config: config dict 
+    :param config: config dict
     :param output_root: dir where output_dir shall be created
     :param restore_from: if not None, from whence the model should be restored (backend-specific information)
 
     :return: main loop object
     """
-    output_dir = dataset = model = hooks = main_loop = None
+    dataset = model = hooks = main_loop = None
+    output_dir = target_dir
 
-    output_dir = create_output_dir(config=config, output_root=output_root)
+    if not output_dir:
```

Comment:
Not sure if this is the only call of the `create_output_dir`. In any case, putting the whole `use the given name or figure out one and make sure it exists` logic to the already-existing `create_output_dir` function would be imo better and more compact. (And you would not have to think about the first sentence)